extensions/arc/src/ui/dialogs/connectSqlDialog.ts

```diff
@@ -4,31 +4,26 @@
  *--------------------------------------------------------------------------------------------*/
 
 import * as azdata from 'azdata';
-import * as vscode from 'vscode';
 import { Deferred } from '../../common/promise';
-import { createCredentialId } from '../../common/utils';
-import { credentialNamespace } from '../../constants';
 import * as loc from '../../localizedConstants';
-import { ControllerModel } from '../../models/controllerModel';
-import { MiaaModel } from '../../models/miaaModel';
 import { InitializingComponent } from '../components/initializingComponent';
 
-export class ConnectToSqlDialog extends InitializingComponent {
-	private modelBuilder!: azdata.ModelBuilder;
+export abstract class ConnectToSqlDialog extends InitializingComponent {
+	protected modelBuilder!: azdata.ModelBuilder;
 
-	private serverNameInputBox!: azdata.InputBoxComponent;
-	private usernameInputBox!: azdata.InputBoxComponent;
-	private passwordInputBox!: azdata.InputBoxComponent;
-	private rememberPwCheckBox!: azdata.CheckBoxComponent;
+	protected serverNameInputBox!: azdata.InputBoxComponent;
+	protected usernameInputBox!: azdata.InputBoxComponent;
+	protected passwordInputBox!: azdata.InputBoxComponent;
+	protected rememberPwCheckBox!: azdata.CheckBoxComponent;
 
-	private _completionPromise = new Deferred<azdata.IConnectionProfile | undefined>();
+	protected _completionPromise = new Deferred<azdata.IConnectionProfile | undefined>();
 
-	constructor(private _controllerModel: ControllerModel, private _miaaModel: MiaaModel) {
+	constructor() {
```

Comment:
Have this be 

`constructor(protected resourceModel) {`

and then have the superclasses pass in their respective models.